lefthook.yml

```diff
@@ -0,0 +1,22 @@
+pre-push:
+  commands:
+    coverage:
+      run: |
+        npm run build &&
+        npm run test:coverage
+
+pre-commit:
+  parallel: true
+  commands:
+    format:
+      glob: '*.{js,ts,md,yml}'
+      run: npm run fmt:check
+    license:
+      run: npm run test:license
+    audit:
+      run: |
+        npm run build &&
+        (run test:audit &
+        npm run test:circular &
+        npm run test:size &)
+        wait
```

Comment:
```yml
pre-push:
  parallel: true
  commands:
    coverage:
      run: npm run test:coverage
    audit:
      run: run test:audit
pre-commit:
  parallel: true
  commands:
    format:
      glob: '*.{js,ts,md,yml}'
      run: npm run fmt:check
    license:
      run: npm run test:license
    size:
      run: npm run test:size
    circular:
      run: npm run test:circular
```

Reply:
I did this in an intermediate iteration, but `license`, `size`, `circular`, `audit` and `coverage` do not work without build.
Do we expect a contributor to build the project himself?

Reply:
We do.

Reply:
Apply your suggest